frontend/src/container/Favorites.jsx

```diff
@@ -0,0 +1,292 @@
+          />
+          &nbsp;
+          {props.t('content is not available')}
+        </span>
+      </div>
+      {props.children}
+    </ListItemWrapper>
+  )
+})
+
+export class Favorites extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      contentCommentsCountList: [],
+      contentBreadcrumbsList: []
+    }
+
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+  }
+
+  handleAllAppChangeLanguage = () => {
+    this.setHeadTitle()
+    this.buildBreadcrumbs()
+  }
+
+  setHeadTitle = () => {
+    const { props } = this
+    const headTitle = buildHeadTitle([props.t('Favorites')])
+    props.dispatch(setHeadTitle(headTitle))
+  }
+
+  buildBreadcrumbs = () => {
+    const { props } = this
+
+    props.dispatch(setBreadcrumbs([{
+      link: PAGE.FAVORITES,
+      type: BREADCRUMBS_TYPE.CORE,
+      label: props.t('Favorites'),
+      isALink: true
+    }]))
+  }
+
+  componentDidMount () {
+    this.setHeadTitle()
+    this.buildBreadcrumbs()
+    this.loadFavoriteList()
+  }
+
+  loadFavoriteList = async () => {
+    const { props } = this
+
+    const fetchFavoriteList = await props.dispatch(
+      getFavoriteContentList(props.user.userId)
+    )
+
+    if (!fetchFavoriteList.ok) {
+      props.dispatch(newFlashMessage(props.t('An error has happened while fetching favorites'), 'warning'))
+      return
+    }
+    const favoriteList = fetchFavoriteList.json.items
+    // Get comments (for their count in info)
+    const commentsFetchList = favoriteList.map(async favorite => {
+      if (!favorite.content) return null
+      // NOTE - S.G. - 2021-04-01 - here we have the favorite as returned by the backend
+      // hence the snake-case properties
+      const response = await getContentComment(
+        FETCH_CONFIG.apiUrl,
+        favorite.content.workspace_id,
+        favorite.content_id
+      )
+      if (!response.ok) return null
+      return (await response.json()).length
+    })
+    const contentCommentsCountList = await Promise.all(commentsFetchList)
+
+    // Get the contents' paths (for breadcrumbs)
+    const contentBreadcrumbsFetchList = favoriteList.map(async favorite => {
+      if (!favorite.content) return null
+      // NOTE - S.G. - 2021-04-01 - here we have the favorite as returned by the backend
+      // hence the snake-case properties
+      const response = await getContentPath(
+        FETCH_CONFIG.apiUrl,
+        favorite.content.workspace_id,
+        favorite.content_id
+      )
+      if (!response.ok) return []
+
+      const workspace = props.workspaceList.find(ws => ws.id === favorite.content.workspace_id)
+
+      return [{ label: workspace.label }].concat((await response.json()).items)
+    })
+    const contentBreadcrumbsList = await Promise.all(contentBreadcrumbsFetchList)
+
+    this.setState({ contentCommentsCountList, contentBreadcrumbsList })
```

Comment:
Yes this would be feasible. I did it this way to keep the favorite objects in redux more like they are in the HTTP API. The frontend has many places where we store a "slightly" changed object compared to the HTTP API. This fact (coupled with the lack of typing) makes things harder to understand I believe.

Reply:
I agree

Reply:
this is more of an implementation issue than an architecture issue
reducers are data for the web app, it doesn't have to be closely sync to the api, it just usually look like it because most of the data are from the api

Reply:
I agree that the data stored in redux is not necessarily directly linked to HTTP APIs but like you wrote most of the time (90% ?) it is the case in Tracim's code.
Also those two additional data (comments count and breadcrumbs) is very closely linked to what we want to display in this page. I can see the favorite list be re-used in other parts of the application (for example to have a add/remove favorite button in other lists), much less this page-specific data.